src/Aspire.Hosting.AWS/Provisioning/Provisioners/AWSProvisioner.cs

```diff
@@ -0,0 +1,167 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Lifecycle;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Aspire.Hosting.AWS.Provisioning;
+
+internal sealed class AWSProvisioner(
```

Comment:
Keep in mind this PR that is approved and just waiting merged. Since the code that PR got moved it will need to be corporated somewhere most likely here. https://github.com/dotnet/aspire/pull/4106

Reply:
We need to coordinate the merge... happy to wait for this PR to happen first or I can just pull in the changes and fix it in the merge conflict. Nevertheless I need to think about how to adopt this for CDK as a construct can have many AWS resources. Or it just both...

Reply:
I see the PR only creates a console URL for the CloudFormation stack. This will automatically work for CDK as well.

Reply:
I'll merge this PR after the #4106 and solve the merge conflicts. This way the credits remain with @djonser